vcd/datasource_vcd_ip_space.go

```diff
@@ -0,0 +1,151 @@
+				Description: "Name of IP space",
+			},
+			"type": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Type of IP space",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Description of IP space",
+			},
+			"internal_scope": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "A set of up internal scope IPs in CIDR format",
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"ip_range_quota": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "IP range quota",
+			},
+			"ip_range": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "IP ranges",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"start_address": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Start address of the IP range",
+						},
+						"end_address": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "End address of the IP range",
+						},
+					},
+				},
+			},
+			"ip_prefix": {
+				Type:        schema.TypeSet,
+				Computed:    true,
+				Description: "IP prefixes",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"prefix": {
+							Type:        schema.TypeSet,
+							Computed:    true,
+							Description: "IP Prefix",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"first_ip": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "First IP",
+									},
+									"prefix_length": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Prefix length",
+									},
+									"prefix_count": {
+										Type:        schema.TypeString,
+										Computed:    true,
+										Description: "Prefix count",
+									},
+								},
+							},
+						},
+						"default_quota": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "Floating IP quota",
+						},
+					},
+				},
+			},
+			"external_scope": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "External scope in CIDR format",
+			},
+			"route_advertisement_enabled": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Flag exposing if route advertisement is enabled",
+			},
+		},
+	}
+}
+
+func datasourceVcdIpSpaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+	log.Printf("[TRACE] IP Space datasource read initiated")
+
+	orgId := d.Get("org_id").(string)
+	ipSpaceName := d.Get("name").(string)
+
+	var ipSpace *govcd.IpSpace
+	var err error
+
+	if orgId != "" { // in case org_id is provided (PRIVATE IP Space)
+		ipSpace, err = vcdClient.GetIpSpaceByNameAndOrgId(ipSpaceName, orgId)
```

Comment:
So, if I understood, it is possible to have two IP Spaces with same name but different org if they're `PRIVATE`, but not if they're SHARED or PUBLIC, right?

Reply:
Yes. Exactly